politeiawww/backend.go

```diff
@@ -106,6 +106,24 @@ func decodeBackendProposalMetadata(payload []byte) (*BackendProposalMetadata, er
 	return &md, nil
 }
 
+// Compare supplied public key against the one stored in the user database
+// It will return the curent active public key if there are no errors
+func checkPublicKey(user *database.User, pk string) ([identity.PublicKeySize]byte, error) {
+	id, ok := database.ActiveIdentity(user.Identities)
+	if !ok {
+		return [identity.PublicKeySize]byte{}, www.UserError{
```

Comment:
Returning `nil` in this function for errors would be preferable to creating an empty array.

Reply:
Try that before. Go compiler throw an error. Unlike C/C++ (which I am more familiar with), Go doesn't seem to allow NULL for a fixed size array.

Reply:
I don't think the array returned should be fixed size anyway because `checkSig` just needs the slice: https://github.com/AlanL1/politeia/blob/8a903d1bae77916fea12a21993e2e5179eda6a56/politeiawww/backend.go#L142